app/src/main/java/com/pvp/app/ui/screen/task/DatePicker.kt

```diff
@@ -0,0 +1,392 @@
+                range = 0..23,
+                selectedValue = selectedDateTime.hour,
+                onValueChange = { hour ->
+                    dateTime = dateTime.withHour(hour)
+                    onDateTimeChanged(dateTime)
+                }
+            )
+        }
+
+        Text(":", style = MaterialTheme.typography.headlineMedium)
+
+        Box(modifier = Modifier.weight(1f)) {
+            TimePickerColumn(
+                range = 0..59,
+                selectedValue = selectedDateTime.minute,
+                onValueChange = { minute ->
+                    dateTime = dateTime.withMinute(minute)
+                    onDateTimeChanged(dateTime)
+                }
+            )
+        }
+    }
+}
+
+@Composable
+fun TimePickerColumn(
+    range: IntRange,
+    selectedValue: Int,
+    onValueChange: (Int) -> Unit
+) {
+    val visibleItemCount = 3
+    val itemCount = range.count()
+    val extendedRange = (range.first - itemCount)..(range.last + itemCount)
+    val initialIndex = itemCount + selectedValue - (visibleItemCount / 2)
+
+    val listState = rememberLazyListState(initialFirstVisibleItemIndex = initialIndex)
+    LaunchedEffect(key1 = selectedValue) {
+        val scrollToIndex = itemCount + selectedValue - (visibleItemCount / 2)
+        listState.scrollToItem(scrollToIndex)
+    }
+
+    LaunchedEffect(key1 = listState) {
+        snapshotFlow { listState.firstVisibleItemIndex }
+            .map { firstVisibleItemIndex ->
+                val correctedIndex = (firstVisibleItemIndex % itemCount) + range.first
+                correctedIndex + (visibleItemCount / 2)
+            }
+            .distinctUntilChanged()
+            .collect { centeredValue ->
+                if (centeredValue in range) {
+                    onValueChange(centeredValue)
+                } else if (centeredValue == range.last + 1) {
+                    // After implementing infinite scroll, first value registers as last + 1.
+                    // Might be better way to address this, but works ig ¯\_(ツ)_/¯
+                    onValueChange(range.first)
+                }
+            }
+    }
+
+    Box(modifier = Modifier.height(50.dp * visibleItemCount)) {
+        LazyColumn(
+            state = listState,
+            modifier = Modifier.fillMaxWidth()
+        ) {
+            items(extendedRange.toList()) { value ->
+                val displayValue = when {
+                    value < range.first -> value + itemCount
+                    value > range.last -> value - itemCount
+                    else -> value
+                }
+                if (displayValue in range) {
+                    TimePickerItem(
+                        value = displayValue,
+                        isSelected = displayValue == selectedValue
+                    ) {}
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun TimePickerItem(value: Int, isSelected: Boolean, onItemSelected: () -> Unit) {
```

Comment:
Multiple parameters could be split into multiple lines.
If `onItemSelected` is not really needed for this component to work, it then can contain a default value `= {}`